test/enigma_test.rb

```diff
@@ -35,12 +35,62 @@ def test_it_can_encrypt
   assert_equal expected, @enigma.encrypt("hello world", "02715", "040895")
   end
 
-  def test_it_can_decrypt
+  def test_it_can_encrypt_with_given_key_and_current_date
+    Time.stubs(:now).returns(Time.new(1995, 04, 8))
+    expected = {
+                encryption: "keder ohulw",
+                key: "02715",
+                date: "040895"
+                }
+  assert_equal expected, @enigma.encrypt("hello world", "02715")
+  end
+
+  def test_it_can_encrypt_with_random_key_and_current_date
+    Time.stubs(:now).returns(Time.new(1995, 04, 8))
+    @enigma.stubs(:generate_number).returns(2715)
+    expected = {
+                encryption: "keder ohulw",
+                key: "02715",
+                date: "040895"
+                }
+  assert_equal expected, @enigma.encrypt("hello world")
+  end
+
+  def test_it_can_decrypt_with_given_date_and_key
     expected = {
                 decryption: "hello world",
                 key: "02715",
                 date: "040895"
                 }
     assert_equal expected, @enigma.decrypt("keder ohulw", "02715", "040895")
   end
+
+  def test_it_can_decrypt_with_given_key_and_current_date
+    Time.stubs(:now).returns(Time.new(1995, 04, 8))
+    expected = {
+                decryption: "hello world",
+                key: "02715",
+                date: "040895"
+                }
+    assert_equal expected, @enigma.decrypt("keder ohulw", "02715")
+  end
+
+  def test_it_can_get_current_date
+    Time.stubs(:now).returns(Time.new(2020, 04, 15))
+    assert_equal "041520", @enigma.current_date
+  end
+
+  def test_it_can_get_random_number
+    number = @enigma.generate_number
+    assert_instance_of Integer, number
+  end
```

Comment:
This is random, so we are just testing that the number exists and that it is an integer. I put it into it's own method to make it easier to test and easier to read.